lib/Backends/OpenCL/Transforms.cpp

```diff
@@ -65,6 +65,14 @@ bool OCLBackend::transformPostLowering(Function *F,
         continue;
       }
 
+      // We need to replace both of the MaxPool results with their NCHW
+      // counterpart in order to get rid of the old node. There appears to be a
+      // bug in the implementation of the result(1) portion of the optimized
+      // kernel, if getArgmax got users - bail.
+      if (PMN->getArgmax().getNumUsers() > 0) {
+        continue;
```

Comment:
If we do not call convertMaxPoolToNCHWPool here, what happens at runtime? Can this operation be executed by the OpenCL backend at all? I.e. is using a non-NCHW layout an option at all?

Reply:
Just like other OpenCL kernels, we have multiple implementations. an 'optimized' one (NCHW) and an unoptimized one (NHWC) in this case. so, if we do not convert, then in this case we will have the reference implementation that'll give lower performance but maintain correctness.